core/logger/logger.go

```diff
@@ -85,10 +90,34 @@ func CreateProductionLogger(
 	config.Level.SetLevel(lvl)
 
 	zl, err := config.Build(zap.AddCallerSkip(1))
+
 	if err != nil {
 		log.Fatal(err)
 	}
-	return &Logger{
-		SugaredLogger: zl.Sugar(),
+	return CreateLogger(zl.Sugar())
+}
+
+func (l *Logger) InitServiceLevelLogger(dir string, serviceName string, jsonConsole bool, toDisk bool, logLevel string) (*zap.SugaredLogger, error) {
```

Comment:
The issue i was running into was that an atomic level seems to only be created per logger, and each child logger seems to inherit the atomic level. This ex is pulled from the [zap docs](https://pkg.go.dev/go.uber.org/zap#NewAtomicLevel):  

`func main() {
	atom := zap.NewAtomicLevel()

	// To keep the example deterministic, disable timestamps in the output.
	encoderCfg := zap.NewProductionEncoderConfig()
	encoderCfg.TimeKey = ""

	logger := zap.New(zapcore.NewCore(
		zapcore.NewJSONEncoder(encoderCfg),
		zapcore.Lock(os.Stdout),
		atom,
	))
	defer logger.Sync()

	logger.Info("info logging enabled")

	atom.SetLevel(zap.ErrorLevel)
	logger.Info("info logging disabled")
}`

We use a similar pattern in the global logger as we create a new logger whenever a logging level is changed. 

I went with creating a new logger per service so we could control each logger with separate levels as it seems in order to have separate logging levels per logging tree, we would need to have separate atomic levels. 

I'll look a bit more if theres another option here but i'm not sure we could set separate atomic levels within one logger/logging tree

Reply:
ugh yeah looks painful https://github.com/uber-go/zap/issues/591. Ok we can leave as is but I think should be a stand alone function then not a method since it doesnt use/depend on the global logger